src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/logging/ApplicationEventsEmitter.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerdeletionservice.logging
+
+import com.microsoft.applicationinsights.TelemetryClient
+import org.springframework.stereotype.Component
+import org.springframework.transaction.event.TransactionalEventListener
+
+@Component
+class ApplicationInsightsEventEmitter(
+  val telemetryClient: TelemetryClient
+) {
+
+  @TransactionalEventListener
```

Comment:
By default this is set to `AFTER_COMMIT`.

`Handle the event after the commit has completed successfully`